### PR TITLE
Implements SLKeyboard protocol for custom keyboards

### PR DESCRIPTION
**Pull request by j-mutter**
With iOS 7 SDK, UIKit now exposes [UIInputView](https://developer.apple.com/library/iOS/documentation/UIKit/Reference/UIInputView_class/Reference/Reference.html) to allow custom inputs/keyboards to be created.

This PR provides an `SLKeyboard` protocol for such views, refactors the original `SLKeyboard` class to implement the protocol methods, and updates `SLTextField` and `SLTextView` to use different keyboards as needed.

### CODE REVIEW

**Comment by wearhere** [comment]
This is neat @j-mutter! Thanks and sorry for the delay in reviewing--I'm slowly getting back into Subliminal after a hefty release of Inkling right before the holidays. I just fixed some things in the build process--would you mind rebasing on top of `master` so that the tests will run against this before I merge?

File: Sources/Classes/UIAutomation/User Interface Elements/SLKeyboard.h [hunk 1]
```diff
@@ -23,27 +23,21 @@
...
 #import "SLButton.h"
 
-/**
- The SLKeyboard allows you to test whether your application's keyboard 
- is visible, and type strings.
-
- To tap individual keys on the keyboard, use SLKeyboardKey.
- */
-@interface SLKeyboard : SLStaticElement
+@protocol SLKeyboard <NSObject>
```
Comment (wearhere) [inline Sources/Classes/UIAutomation/User Interface Elements/SLKeyboard.h:36]: This is an elegant way to support different keyboards. Before your request I hadn't heard of `UIInputView`--would you mind describing it/the rationale here in documentation for the `SLKeyboard` protocol?

**Comment by j-mutter** [comment]
@wearhere Done and done :) Sorry for the delay

**Comment by wearhere** [comment]
Looks like everything passed except for some documentation errors--[missing parameters](https://travis-ci.org/inkling/Subliminal/jobs/18599112#L469) on the new methods added to `SLTextField` and `SLTextView`. Can you fix @j-mutter?

**Comment by j-mutter** [comment]
Travis is :green_heart: now :)

**Comment by wearhere** [comment]
:+1: Thanks again!

**Pull request merged**